node/src/neighborhood/mod.rs

```diff
@@ -1448,6 +1458,79 @@ impl Neighborhood {
         undesirability + node_undesirability
     }
 
+    fn handle_exit_location_message(&mut self, message: UiSetExitLocationRequest) {
+        let exit_locations_by_priority: HashMap<usize, Vec<String>> = message
```

Comment:
It doesn't look like you're ever looking up a list of countries by priority. That means you don't really need the hashiness of the `HashMap`; you could probably replace this with a `Vec<(usize, Vec<String>)>`. Then the contents would always be in the same order.